dask/array/core.py

```diff
@@ -2488,7 +2504,7 @@ def insert_to_ooc(out, arr, lock=True, region=None):
     if lock is True:
         lock = Lock()
 
-    def store(x, index, lock, region):
+    def store(out, x, index, lock, region):
```

Comment:
Oh interesting.  It looks like previously we were adding `out` into the function as a closure.  This definitely wasn't desired.  Thanks for cleaning this up.